build: Add support for linking to shared falcosecurity libraries.

apteryks opened the pull request:
* cmake/modules/falcosecurity-libs.cmake (USE_BUNDLED_FALCOSECURITY_LIBS): New option.  Look for libsinsp via pkg-config unless it's enabled.
* userspace/sysdig/CMakeLists.txt: Adjust link directives accordingly. Remove extraneous zlib include.

apteryks committed:
build: Add support for linking to shared falcosecurity libraries.

* cmake/modules/falcosecurity-libs.cmake (USE_BUNDLED_FALCOSECURITY_LIBS):
New option.  Look for libsinsp via pkg-config unless it's enabled.
* userspace/sysdig/CMakeLists.txt: Adjust link directives accordingly.

Signed-off-by: Maxim Cournoyer <maxim.cournoyer@gmail.com>

therealbobo commented on userspace/sysdig/CMakeLists.txt:
```diff
@@ -108,7 +107,7 @@ if(NOT WIN32)
 
 	target_link_libraries(sysdig
-		sinsp
+		$<IF:$<BOOL:${USE_BUNDLED_FALCOSECURITY_LIBS}>,sinsp,PkgConfig::LIBSINSP>
```
```suggestion
		"${SINSP_LIB}"
```
and define it with a simple
```
if(USE_BUNDLED_FALCOSECURITY_LIBS)
    set(SINSP_LIB sinsp)
else()
    set(SINSP_LIB PkgConfig::LIBSINSP)
endif()
```

I think that an approach like this will ensure a better readability of the cmake. WDYT?

apteryks replied:
Done, thanks for the review.

apteryks commented:
Note that this depends on https://github.com/falcosecurity/libs/pull/1842 to work.

apteryks commented:
Not stale (that's an aggressive bot!).

apteryks commented:
Hi, I'll try to update it this week.

apteryks commented:
So, looking back at this, are there actual problems preventing this from being merged?  I see a windows and macos-13 build failing, but maybe these are flukes/known issues?

apteryks commented:
PS: that bot really gets on my nerves

therealbobo commented:
seems good to me! I build is currently broken and I have an ongoing PR to fix it. Once is merged I'll relaunch le build on this PR.

therealbobo commented:
@Apteryks do you mind rebasing before merging?

therealbobo commented on userspace/sysdig/CMakeLists.txt:
```diff
@@ -20,7 +20,6 @@ if(NOT WIN32)
 	include(ncurses)
 endif() # NOT WIN32
 
-include(zlib)
```
Why zlib has been removed?

apteryks replied:
It's now (more accurately, will after https://github.com/falcosecurity/libs/pull/1842 is merged) required by the pkg-config files of falcosecurity-libs, so it became redundant.

apteryks replied:
But maybe it's still useful for odd platforms lacking pkg-config or wanting to build the bundled zlib?  I don't believe this exists given this is a Linux-specific project (right?) and every distro on earth has a zlib package.

apteryks replied:
If my reasoning is correct, we could then delete the `cmake/modules/zlib.cmake` remnant.

therealbobo replied:
> Linux-specific project

Not exactly: there are some users that uses sysdig to analyze scap captures :/

apteryks replied:
I've left it in; it won't hurt.

apteryks committed:
Update falcosecurity-libs to latest (0.20.0).

* cmake/modules/falcosecurity-libs.cmake: Require falcosecurity-libs
0.20.0.
* userspace/chisel/chisel_api.cpp (lua_cbacks::get_thread_table_int):
Adjust for falcosecurity-libs commit
417e523399796d7290a479fd55988e62e80fc97e ("drop user and group infos
embedded in threadinfo."); this is a follow-up that simplifies
04f54e51c a bit.

Fixes: #2132
Signed-off-by: Maxim Cournoyer <maxim.cournoyer@gmail.com>

apteryks commented:
> @Apteryks do you mind rebasing before merging?

Done!
